aws_crt_common_ws/aws-crt-c-flags/src/lib.rs

```diff
@@ -0,0 +1,469 @@
+                .add_private_cflag("-pedantic")
+                .add_private_cflag("-Wno-long-long")
+                .add_private_cflag("-fPIC");
+        }
+
+        if self.build_toolchain.is_flag_supported("-Wgnu").is_ok() {
+            // -Wgnu-zero-variadic-macro-arguments results in a lot of false positives
+            self.add_private_cflag("-Wgnu")
+                .add_private_cflag("-Wno-gnu-zero-variadic-macro-arguments");
+
+            if self
+                .try_compile(
+                    " #include <netinet/in.h>
+            int main() {
+            uint32_t x = 0;
+            x = htonl(x);
+            return (int)x;
+            }",
+                )
+                .is_err()
+            {
+                self.add_private_cflag("-Wno-gnu-statement-expression");
+            }
+        }
+
+        for pub_flag in &self.public_cflags {
+            self.build_toolchain.flag_if_supported(pub_flag.as_str());
+        }
+
+        for priv_flag in &self.private_cflags {
+            self.build_toolchain.flag_if_supported(priv_flag.as_str());
+        }
+
+        for pub_define in &self.public_defines {
+            self.build_toolchain
+                .define(pub_define.0.as_str(), pub_define.1.as_str());
+        }
+
+        for priv_define in &self.private_defines {
+            self.build_toolchain
+                .define(priv_define.0.as_str(), priv_define.1.as_str());
+        }
+
+        for include in &self.include_dirs {
+            self.build_toolchain.include(include);
+        }
+
+        for module in &self.crt_module_deps {
+            for pub_flag in &module.public_cflags {
+                self.build_toolchain.flag(pub_flag.as_str());
+            }
+
+            for pub_define in &self.public_defines {
+                self.build_toolchain
+                    .define(pub_define.0.as_str(), pub_define.1.as_str());
+            }
+
+            for include in &self.include_dirs {
+                self.build_toolchain.include(include);
+            }
+        }
+
+        if self.shared_lib {
+            self.build_toolchain.shared_flag(true);
+        }
+    }
+
+    /// Executes the build and if successful stores this object in the environment for the next crate to use.
+    pub fn run_build(&mut self) {
+        self.load_to_build();
+        print!("{}", serde_json::to_string(self).unwrap().as_str());
+        self.build_toolchain.compile(self.lib_name.as_str());
+
+        if self.linker_path.is_some() {
+            println!(
+                "cargo:rustc-link-search={}",
+                self.linker_path.as_ref().unwrap().to_str().unwrap()
+            );
+        }
+
+        for link_flag in &self.link_targets {
+            println!("cargo:rustc-link-lib={}", link_flag)
+        }
+
+        let module_name_cpy = self.crt_module_name.clone();
+        env::set_var(
+            "CRT_MODULE_".to_owned() + module_name_cpy.as_str() + "BUILD_CFG",
```

Comment:
This has the same problem as the string @rcoh commented on above, use `"{}"`

Reply:
this is slightly different (although I'd use the format macro here personally). The problem I mentioned above is specific to `panic!` which will only accept static strings as the first argument in Rust 2021